src/main/java/com/tabnine/binary/requests/autocomplete/CompletionPostprocess.kt

```diff
@@ -0,0 +1,48 @@
+package com.tabnine.binary.requests.autocomplete
+
+fun postprocess(request: AutocompleteRequest, result: AutocompleteResponse, tabSize: Int) {
+    result.results.forEach { it.new_prefix = it.new_prefix.replace("\t", " ".repeat(tabSize)) }
+
+    val requestIndentation = lastLineIndentation(request.before.replace("\t", " ".repeat(tabSize))) ?: return
+    if (requestIndentation == 0) return
+
+    val regex = constructRegex(requestIndentation)
+    result.results.forEach { entry ->
+        val calculateTrimmingIndex = calculateTrimmingIndex(entry.new_prefix, regex)
+        calculateTrimmingIndex?.let { entry.new_prefix = entry.new_prefix.take(it) }
+    }
+}
+
+/**
+ * Finds the amount of spaces or tabs in the last line of the given `value`,
+ * returning `null` if `value` has no newlines, or the last line is not whitespaces only.
+ */
+fun lastLineIndentation(value: String): Int? {
+    try {
+        val lastLine = value.lines().last()
```

Comment:
Is there a way to scan from the end for the 1st linebreak? Won't it be faster for larger texts?

Reply:
good point - can use somthing like:

```kotlin
value.substring(value.lastIndexOf("\n"))
```